src/main/resources/static/js/subscription.js

```diff
@@ -739,85 +739,95 @@ jQuery(document).ready(function () {
 
     // /Start ## Save Subscription ##########################################
     $('div.modal-footer').on('click', 'button.save_record', function (event) {
+        var error = false;
         event.stopPropagation();
         event.preventDefault();
         var notificationMessageKeyValuesArray = vm.subscription()[0].notificationMessageKeyValues();
         if (!vm.formpostkeyvaluepairs()) {
             notificationMessageKeyValuesArray[0].formkey = ""; // OBS must be empty when NOT using REST POST Form key/value pairs
         }
 
-
+        $('.hiddenField').hide();
         //START: Make sure all datatables field has a value
+        // Validate SubscriptionName field
+        if (vm.subscription()[0].subscriptionName() == "") {
+            window.logMessages("Error: SubscriptionName field must have a value");
+            $('#noNameGiven').show();
+            error = true;
+        }
         if (!(/[a-z]|[A-Z]|[0-9]|[\_]/.test(String(vm.subscription()[0].subscriptionName()).slice(-1)))) {
             window.logMessages("Only numbers,letters and underscore is valid to type in subscriptionName field.");
-            return;
+            $('#invalidLetters').show();
+            error = true;
         }
 
-        //Currently a free-form field
-        /*
-            if (!(/[a-z]|[A-Z]|[0-9]|[\:\/\.]/.test(String(vm.subscription()[0].notificationMeta()).slice(-1)))) {
-                window.logMessages("Only numbers and letters is valid to type in notificationMeta field.");
-                return;
-            }
-        */
-
-        if (vm.subscription()[0].subscriptionName() == "") {
-            window.logMessages("Error: SubscriptionName field must have a value");
-            return;
-        }
+        // Validate notificationType field
         if (vm.subscription()[0].notificationType() == null) {
             window.logMessages("Error: notificationType field must boolean a value");
```

Comment:
This log does not make sense. Maybe have "Error: notificationType must have a value"?

Reply:
Maybe to remove those messages since it error now appear in at the field where the values was added.

Reply:
I think it's fine to keep both (as long as the information is the same) to clearly highlight that something is wrong.